### DISCUSSION TITLE
Nbgrader on a new server

### DISCUSSION THREAD

**Discussion by Mad31** [jupyter/nbgrader#1891] [Q&A]
Hi,
i try to install formgader on my new server in a docker.

I can't access to formgrader page , the server refuse the connexion !
![image](https://github.com/jupyter/nbgrader/assets/7804553/c6e0267c-6aa3-4157-be34-fd35de0a91f4)

thx for helping, 

my configuration below : 

adminjh@5ce4f66a9e1b:~$ jupyter --version
Selected Jupyter core packages...
IPython          : 8.25.0
ipykernel        : 6.29.4
ipywidgets       : 8.1.3
jupyter_client   : 8.6.2
jupyter_core     : 5.7.2
jupyter_server   : 2.14.1
jupyterlab       : 4.2.2
nbclient         : 0.10.0
nbconvert        : 7.16.4
nbformat         : 5.10.4
notebook         : 7.2.1
qtconsole        : 5.5.2
traitlets        : 5.14.3

JupyterLab v4.2.2
/usr/local/share/jupyter/labextensions
        jupyterlab_pygments v0.3.0 enabled OK (python, jupyterlab_pygments)
        jupyter-py5 v0.1.0 enabled OK (python, py5jupyter)
        jupyterlab-myst v2.4.2 enabled OK (python, jupyterlab_myst)
        ipycanvas v0.13.2 enabled OK
        nbgrader v0.9.2 enabled  X (python, nbgrader)
        @jupyter-notebook/lab-extension v7.2.1 enabled OK
        @jupyter-widgets/jupyterlab-manager v5.0.11 enabled OK (python, jupyterlab_widgets)

**Comment by Mad31** [comment]
Remark :
I can do all in command line,but not with the formgrader page that i can't reach.

**Comment by brichet** [comment]
Thanks @Mad31 for opening this.
I'm seeing some issues with the current release of `nbgrader` and `jupyterlab>=4.2.0`.
Can you try downgrading jupyterlab to `4.1.x` ?

Not sure it can be related to your server issue though

**Comment by Mad31** [comment]
No it's the same..
When i open inspector i have : 
Refused to frame 'http://xx.xx.xx.xx:8000/' because an ancestor violates the following Content Security Policy directive: "frame-ancestors 'none'".

**Comment by brichet** [comment]
OK, your are probably running it with Jupyterhub 4.1.x.
I didn't try but there is a workaround here: https://github.com/jupyter/nbgrader/issues/1870

**Comment by Mad31** [comment]
thx for all
adding to jupyter_config.py :
```
c.JupyterHub.tornado_settings = {'headers': {'Content-Security-Policy': "frame-ancestors *;"}}
c.Spawner.args = ['--NotebookApp.tornado_settings={"headers":{"Content-Security-Policy": "frame-ancestors *;"}}']
```
solve the problem
